### ISSUE TITLE
Unable to Decrypt Game Paks (1.5.2)

### ISSUE THREAD

**Issue by Blaze6265** [CutFlame/MCDSaveEdit#129]
The, uh, "short" version is that the editor couldn't find the Game Paks in the usual location that it is in, so I simply put in the path to it, which is in Roaming, not Local, and put the path in. I expected it to work like any other time I used it, but it instead showed an error that it was unable to decrypt the Game Paks. It seems I'm not the only one with this issue, and yes, it's the latest version.

![Screenshot 2022-04-21 103532](https://user-images.githubusercontent.com/104151839/164482886-9de1e75e-1f13-469b-bb89-117f3e21460a.png)

I think it has something to do with an invalid key or something like that, my game is fully updated too. The editor usually works without _other fancy projects_ working with it, so I was confused on what to do. I also have a mod installed there, but the editor also doesn't seem to mind that, so... anyone have tips?

**Comment by LuckyPants12** [comment]
Yeah i have the problem , its likely due to the MCD version being out of date, waiting for the creator to update is probably the only fix

**Comment by nicfanst** [comment]
I believe Mojang might have (?) patched the save editor, as Pak files in the Pak folder are named `pakchunk(id)-WindowsNoEditor.pak`
![image](https://user-images.githubusercontent.com/81713476/164863281-8bd755fa-af32-49dc-b158-ba94e0ade69a.png)

**Comment by Blaze6265** [comment]
I don't really know, I thought that was always there. And here I am, knowing nothing about "`PAKs`" and "keys" and "decrypting" and "`IDs`", and still having the gall to open this issue and talk about it.

**Comment by LuckyPants12** [comment]
> I believe Mojang might have (?) patched the save editor, as Pak files in the Pak folder are named `pakchunk(id)-WindowsNoEditor.pak` ![image](https://user-images.githubusercontent.com/81713476/164863281-8bd755fa-af32-49dc-b158-ba94e0ade69a.png)

I am sure they were always named like that , you can see in the ReamdMe that picture which had simmilar names to the newer .pak files

**Comment by LuckyPants12** [comment]
Ok so i got it working.
You just need to compile the project with the correct AES key as shows in the compile ReadME.
The AES key for the new version is ``0x16D28F03446CB9E15665F11EF5A0D6820A6483C083CA6DF1AE9370D197ACB588``

**Comment by RedPanda814** [comment]
> Ok so i got it working. You just need to compile the project with the correct AES key as shows in the compile ReadME. The AES key for the new version is `0x16D28F03446CB9E15665F11EF5A0D6820A6483C083CA6DF1AE9370D197ACB588`

How do I do that? I'm new to this so I don't get what you mean by compile the project.

**Comment by TeamAzuriteRexi** [comment]
Alright so i have tried to do that how ever i didnt seem to find the "Secret" file to paste the new AES Key in would you mind guiding us in a more detailed way on how to do it

> Ok so i got it working. You just need to compile the project with the correct AES key as shows in the compile ReadME. The AES key for the new version is `0x16D28F03446CB9E15665F11EF5A0D6820A6483C083CA6DF1AE9370D197ACB588`

**Comment by Blaze6265** [comment]
> Ok so i got it working.
> You just need to compile the project with the correct AES key as shows in the compile ReadME.
> The AES key for the new version is `0x16D28F03446CB9E15665F11EF5A0D6820A6483C083CA6DF1AE9370D197ACB588`

So, I’m, that’s very great, but I am unable to figure out what that means. I guess I’ll have to do a bit of digging when I have time.

**Comment by TeamAzuriteRexi** [comment]
> 

Please let me know if you find anything i am able to use

> > Ok so i got it working.
> > You just need to compile the project with the correct AES key as shows in the compile ReadME.
> > The AES key for the new version is `0x16D28F03446CB9E15665F11EF5A0D6820A6483C083CA6DF1AE9370D197ACB588`
> 
> So, that’s very great, but I am unable to figure out what that means. I guess I’ll have to do a bit of digging when I have time.

**Comment by LuckyPants12** [comment]
I have the working exe uploaded on my fork of this project, just chek my profile and you will find it currently working on making it able to change the AES code without needing to rebuild it with just a simple settings option

**Comment by LuckyPants12** [comment]
> I have the working exe uploaded on my fork of this project, just chek my profile and you will find it currently working on making it able to change the AES code without needing to rebuild it with just a simple settings option

Have uploaded the pre release for the AES key setting, please chek them for any bugs if you decide to use them, you can still use the 1.14.1.0 version only build i have released in the same forked repository.

**Comment by TeamAzuriteRexi** [comment]
Thanks alot 

> > I have the working exe uploaded on my fork of this project, just chek my profile and you will find it currently working on making it able to change the AES code without needing to rebuild it with just a simple settings option
> 
> Have uploaded the pre release for the AES key setting, please chek them for any bugs if you decide to use them, you can still use the 1.14.1.0 version only build i have released in the same forked repository.

**Comment by RedPanda814** [comment]
> I have the working exe uploaded on my fork of this project, just chek my profile and you will find it currently working on making it able to change the AES code without needing to rebuild it with just a simple settings option

I don't see an exe I only see a zip file

**Comment by Blaze6265** [comment]
> > I have the working exe uploaded on my fork of this project, just chek my profile and you will find it currently working on making it able to change the AES code without needing to rebuild it with just a simple settings option
> 
> I don't see an exe I only see a zip file

For that, you must extract the .zip file, and then enter the folder. You’ll see the .exe application.

**Comment by RedPanda814** [comment]
> > > I have the working exe uploaded on my fork of this project, just chek my profile and you will find it currently working on making it able to change the AES code without needing to rebuild it with just a simple settings option
> > 
> > 
> > I don't see an exe I only see a zip file
> 
> For that, you must extract the .zip file, and then enter the folder. You’ll see the .exe application.
> 
> Or, at least, I think that’s how it works.
![MCDSaveEdit-master 24_04_2022 18_44_21](https://user-images.githubusercontent.com/104303462/164987131-bbf4db27-6a71-4326-920a-2f538916f0a4.png)

**Comment by LuckyPants12** [comment]
> > > > I have the working exe uploaded on my fork of this project, just chek my profile and you will find it currently working on making it able to change the AES code without needing to rebuild it with just a simple settings option
> > > 
> > > 
> > > I don't see an exe I only see a zip file
> > 
> > 
> > For that, you must extract the .zip file, and then enter the folder. You’ll see the .exe application.
> > Or, at least, I think that’s how it works.
> > ![MCDSaveEdit-master 24_04_2022 18_44_21](https://user-images.githubusercontent.com/104303462/164987131-bbf4db27-6a71-4326-920a-2f538916f0a4.png)
> 
> I don't see an executable

you are suppose to go to the releases page and download the MCDSaveEdit.exe , and download it, either one should work
you just need to manually enter the AES code in the pre release one
<img width="502" alt="image" src="https://user-images.githubusercontent.com/75674528/164988193-0ee5bfff-f080-44a0-8f48-c7e162a45b8b.png">

**Comment by RedPanda814** [comment]
> > > > > I have the working exe uploaded on my fork of this project, just chek my profile and you will find it currently working on making it able to change the AES code without needing to rebuild it with just a simple settings option
> > > > 
> > > > 
> > > > I don't see an exe I only see a zip file
> > > 
> > > 
> > > For that, you must extract the .zip file, and then enter the folder. You’ll see the .exe application.
> > > Or, at least, I think that’s how it works.
> > > ![MCDSaveEdit-master 24_04_2022 18_44_21](https://user-images.githubusercontent.com/104303462/164987131-bbf4db27-6a71-4326-920a-2f538916f0a4.png)
> > 
> > 
> > I don't see an executable
> 
> you are suppose to go to the releases page and download the MCDSaveEdit.exe , and download it, either one should work you just need to manually enter the AES code in the pre release one <img alt="image" width="502" src="https://user-images.githubusercontent.com/75674528/164988193-0ee5bfff-f080-44a0-8f48-c7e162a45b8b.png">

thank you

**Comment by LuckyPants12** [comment]
> > > > > > I have the working exe uploaded on my fork of this project, just chek my profile and you will find it currently working on making it able to change the AES code without needing to rebuild it with just a simple settings option
> > > > > 
> > > > > 
> > > > > I don't see an exe I only see a zip file
> > > > 
> > > > 
> > > > For that, you must extract the .zip file, and then enter the folder. You’ll see the .exe application.
> > > > Or, at least, I think that’s how it works.
> > > > ![MCDSaveEdit-master 24_04_2022 18_44_21](https://user-images.githubusercontent.com/104303462/164987131-bbf4db27-6a71-4326-920a-2f538916f0a4.png)
> > > 
> > > 
> > > I don't see an executable
> > 
> > 
> > you are suppose to go to the releases page and download the MCDSaveEdit.exe , and download it, either one should work you just need to manually enter the AES code in the pre release one <img alt="image" width="502" src="https://user-images.githubusercontent.com/75674528/164988193-0ee5bfff-f080-44a0-8f48-c7e162a45b8b.png">
> 
> thank you

Just need to make sure if the pre release also works for others, the pre release had a manual AES code adder and it means you dont need to compile anything anymore, just need to get the AES code after every update and put it in the settings page i made on the top, close the  app and reopen it, volla it works with the newest version now, it can also work with every other version like this , AES codes can be found at https://discord.gg/CsjHbQkNGr, type ?key in the #dungeons channel

**Comment by CutFlame** [comment]
Every time MCD publishes an update they change the AES key to decrypt their game content and most of the time they change some things in the structure of the save files as well. It takes time to reverse engineer these changes and then make MCDSaveEdit accommodate so that there is no data loss when it saves. This is one of the main reasons I have not supported a custom AES key input in MCDSaveEdit. The recent addition of the storage chest illustrates this issue. If you used @LuckyPants12 build (nice work BTW) and put in the new AES key and saved your changes to your character, you may notice that you lost all your items stored in your storage chest. So, yes, you can use that feature (and I might even put it in my version) but you do so at your own risk and I ask that you not blame me for your data loss.

TL;DR - I may put in a feature to allow input of a custom AES key but, without updating MCDSaveEdit, it could cause character data loss and I won't recommend it

**Comment by ZUZ3** [comment]
> Every time MCD publishes an update they change the AES key to decrypt their game content and most of the time they change some things in the structure of the save files as well. It takes time to reverse engineer these changes and then make MCDSaveEdit accommodate so that there is no data loss when it saves. This is one of the main reasons I have not supported a custom AES key input in MCDSaveEdit. The recent addition of the storage chest illustrates this issue. If you used @LuckyPants12 build (nice work BTW) and put in the new AES key and saved your changes to your character, you may notice that you lost all your items stored in your storage chest. So, yes, you can use that feature (and I might even put it in my version) but you do so at your own risk and I ask that you not blame me for your data loss.
> 
> TL;DR - I may put in a feature to allow input of a custom AES key but, without updating MCDSaveEdit, it could cause character data loss and I won't recommend it

Honestly would rather have a fully working version so keep up the amazing work you are doing CutFlame.

**Comment by Blaze6265** [comment]
> Every time MCD publishes an update they change the AES key to decrypt their game content and most of the time they change some things in the structure of the save files as well. It takes time to reverse engineer these changes and then make MCDSaveEdit accommodate so that there is no data loss when it saves. This is one of the main reasons I have not supported a custom AES key input in MCDSaveEdit. The recent addition of the storage chest illustrates this issue. If you used @LuckyPants12 build (nice work BTW) and put in the new AES key and saved your changes to your character, you may notice that you lost all your items stored in your storage chest. So, yes, you can use that feature (and I might even put it in my version) but you do so at your own risk and I ask that you not blame me for your data loss.
> 
> TL;DR - I may put in a feature to allow input of a custom AES key but, without updating MCDSaveEdit, it could cause character data loss and I won't recommend it

I see. That is very amazing, but I can just barely understand what that means. All I'm sure of is that it's very hard to do.
_If only I knew how to do such works of art._

**Comment by LuckyPants12** [comment]
> Every time MCD publishes an update they change the AES key to decrypt their game content and most of the time they change some things in the structure of the save files as well. It takes time to reverse engineer these changes and then make MCDSaveEdit accommodate so that there is no data loss when it saves. This is one of the main reasons I have not supported a custom AES key input in MCDSaveEdit. The recent addition of the storage chest illustrates this issue. If you used @LuckyPants12 build (nice work BTW) and put in the new AES key and saved your changes to your character, you may notice that you lost all your items stored in your storage chest. So, yes, you can use that feature (and I might even put it in my version) but you do so at your own risk and I ask that you not blame me for your data loss.
> 
> TL;DR - I may put in a feature to allow input of a custom AES key but, without updating MCDSaveEdit, it could cause character data loss and I won't recommend it

Thanks , and i love your work btw! Yea the storage chest isnt a thing i considered while recompiling your build, i am not THAT good with coding haha, anyways ill add a warning just in case, or if you want me to delete ill be happy to :)

**Comment by Blaze6265** [comment]
I'm going to now close this issue, being that the 1.5.3 update of the application now fixes my issue.
